packages/codegen/src/inheritance.ts

```diff
@@ -125,7 +135,7 @@ function expandSingleTableInheritance(
         // Make fields as required
         function maybeRequired<T extends { notNull: boolean; fieldName: string }>(field: T): T {
           const config = subTypeFields.find(([name]) => name === field.fieldName)?.[1]!;
-          if (config.stiNotNull) field.notNull = true;
+          if (config.notNull) field.notNull = true;
```

Comment:
Kinda wonder if this `maybeRequired` could go away, now that `setupSubTypeSpecialization` is doing it as well?

Reply:
Thinking more about it, I think that `notNull` should only be read from `entityConfig` and not fall back to `baseConfig` inside `setupSubTypeSpecialization`.  Since STI descendants don't have per type config, it'd never be read from their config so they'd effectively be skipped here.  Meanwhile, it doesn't make sense for CTI to define a `notNull: true` on the base config since they could just make the db level field required in the first place.

Reply:
> I think that notNull should only be read from entityConfig and not fall back to baseConfig inside setupSubTypeSpecialization

Makes sense

> Since STI descendants don't have per type config

I thought we wanted that / were thinking of adding it?

Reply:
> I thought we wanted that / were thinking of adding it?

I mean maybe at some point, but right now we don't have it and we don't need it immediately, so we can punt on this until then?